File: _정답코드/5주차/04_기준_점수_초과_인원_집계/Solution.java
class Solution {
    public int[] solution(int[] numbers, int infimum, int[][] queries) {
        int n = numbers.length;
        int[] prefixCount = new int[n + 1];
        
        for (int i = 0; i < n; i++) {
            prefixCount[i + 1] = prefixCount[i];
            if (numbers[i] > infimum) {
                prefixCount[i + 1]++;
            }
        }
        
        int[] result = new int[queries.length];
        for (int i = 0; i < queries.length; i++) {
            int l = queries[i][0];
            int r = queries[i][1];
            result[i] = prefixCount[r] - prefixCount[l - 1];
        }
        
        return result;
    }
}

File: _정답코드/5주차/04_기준_점수_초과_인원_집계/solution.py
def solution(numbers, infimum, queries):
    n = len(numbers)
    prefixCount = [0] * (n + 1)
    
    for i in range(n):
        prefixCount[i + 1] = prefixCount[i]
        if numbers[i] > infimum:
            prefixCount[i + 1] += 1
    
    result = []
    for i in range(len(queries)):
        l = queries[i][0]
        r = queries[i][1]
        result.append(prefixCount[r] - prefixCount[l - 1])
    
    return result

